rest/route/github.go

```diff
@@ -462,6 +462,13 @@ func (gh *githubHookApi) handleMergeGroupChecksRequested(event *github.MergeGrou
 		"head_sha": event.GetMergeGroup().GetHeadSHA(),
 		"message":  "merge group received",
 	})
+	projectRefs, err := model.FindMergedEnabledProjectRefsByRepoAndBranch(org, repo, branch)
```

Comment:
Rather than checking project refs again here, maybe we could check more intentionally at the place where we're erroring instead. I suggest this because it's a little special-casey to check all project refs here for such a specific problem; we really might not know why, looking back, we're doing this, and it goes against the pattern of leaving that logic for the patch intent.

At the place right now where you check for a [merge queue project](https://github.com/evergreen-ci/evergreen/blob/4c3c042886a42ef7a47cf97f55c8c9617301b782/units/patch_intent.go#L1066), would if you instead just ran this query to find all matching project refs? If none patch, then you know the project is totally untracked, but if projects are returned and none of them have the merge queue enabled (there will be exactly one) you know to return an error back to GitHub.